src/fastapi_poe/base.py

```diff
@@ -182,8 +193,13 @@ async def get_response_with_context(
         response to the Poe servers. This is what gets displayed to the user.
 
         """
-        async for event in self.get_response(request):
-            yield event
+        try:
+            async for event in self.get_response(request):
+                yield event
+        except InsufficientFundError:
+            yield ErrorResponse(
+                allow_retry=True, error_type="insufficient_fund", text=""
```

Comment:
not sure if allow_retry=True is intentional - generally for insufficient_fund its not retryable right? Or is this to allow retry with a higher cap?

Reply:
Yes, I intentionally set to True to allow the user to retry a message with a higher point limit.

Reply:
Oh, thanks for pointing out! When testing both allow_retry = True and allow_retry = False did not trigger a retry but still allowed the user to retry (maybe because of the different error type?).